HPCC-12942 Process standard query responses correctly

rpastrana opened the pull request:
- XmlWriter should allow fully qualified ns (don't append xmlns automatically)
- Query responses should be named appropriately
- Remove redundant response processing code

Signed-off-by: rpastrana rodrigo.pastrana@lexisnexis.com

rpastrana committed:
HPCC-12942 Process standard query responses correctly

- XmlWriter should allow fully qualified ns (don't append xmlns automatically)
- Query responses should be named appropriately
- Remove redundant response processing code

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@afishbeck please review. In particular please consider the change to the xmlwriter ns logic. Also, left commented out code in order to show the actual change. I will remove after your review. Thanks.

afishbeck commented on common/thorhelper/thorxmlwrite.cpp:
```diff
@@ -246,7 +246,10 @@ void CommonXmlWriter::outputXmlns(const char *name, const char *uri)
 {
     StringBuffer fieldname;
     if (!streq(name, "xmlns"))
-        fieldname.append("xmlns:");
```
This change isn't necessary, or actually correct.  Should probably be an exception if there is a colon in the xmlns.. see comment below.

richardkchapman replied:
Has this comment been addressed?

rpastrana replied:
Yes, I was misusing this method. My original change to this method is not necessary.

afishbeck replied:
looks like an extra tab is still added.

afishbeck commented on esp/esdllib/esdl_transformer2.cpp:
```diff
@@ -1317,7 +1317,7 @@ void Esdl2Response::process(Esdl2TransformerContext &ctx, const char *out_name,
                     {
-                        ctx.writer->outputXmlns("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance");
+                        ctx.writer->outputXmlns("xsi", "http://www.w3.org/2001/XMLSchema-instance");
                         ctx.writer->outputXmlns("xsi:schemaLocation", ctx.schemaLocation.str());
```
This is not a namespace, and shouldn't call outputXmlns.  This is an attribute that uses the "xsi"  namespace.

rpastrana committed:
HPCC-12942 Code review changes

- Remove changes added to XMLWriter
- treat xsi:schemalocation as a simple xml attribute

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@afishbeck please review latest change.

afishbeck commented:
@rpastrana looks fine.

rpastrana commented:
@richardkchapman please merge this PR, thanks.

afishbeck commented on esp/esdllib/esdl_transformer2.cpp:
```diff
@@ -1817,7 +1817,30 @@ void Esdl2Transformer::processHPCCResult(IEspContext &ctx, IEsdlDefMethod &mthde
-                        resp->processChildNamedResponse(tctx, restype);
+                        resp->process(tctx, restype);//resp->processChildNamedResponse(tctx, restype);
+                }
+            }*/
```
Also you said you'd remove commented out code after my review :)

rpastrana committed:
HPCC-12942 Second code review changes

- Remove extra tab
- Remove commented out block.
Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
Both issues addressed.
